config/config.go

```diff
@@ -0,0 +1,145 @@
+)
+
+// AppConfig ... Highest order config. Stores
+// all relevant fields necessary for running both proxy
+// & metrics servers.
+type AppConfig struct {
+	EigenDAConfig ProxyConfig
+	MetricsCfg    metrics.Config
+}
+
+func (c AppConfig) Check() error {
+	err := c.EigenDAConfig.Check()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func ReadCLIConfig(ctx *cli.Context) AppConfig {
+	return AppConfig{
+		EigenDAConfig: ReadProxyConfig(ctx),
+		MetricsCfg:    metrics.ReadConfig(ctx),
+	}
+}
+
+// ProxyConfig ... Higher order config which bundles all configs for instrumenting
+// the proxy server with necessary client context
+type ProxyConfig struct {
+	ServerConfig        server.Config
+	EdaV1ClientConfig   clients.EigenDAClientConfig
+	EdaV1VerifierConfig verify.Config
+
+	EdaV2ClientConfig common.V2ClientConfig
+
+	MemstoreConfig memstore.Config
+	StorageConfig  store.Config
+	PutRetries     uint
+
+	MemstoreEnabled bool
+
+	EigenDAV2Enabled bool
+}
+
+type V2ClientConfig struct {
+	DisperserClientCfg clients_v2.DisperserClientConfig
+	PayloadClientCfg   clients_v2.PayloadDisperserConfig
+	RetrievalConfig    clients_v2.RelayPayloadRetrieverConfig
+}
+
+// ReadProxyConfig ... parses the Config from the provided flags or environment variables.
+func ReadProxyConfig(ctx *cli.Context) ProxyConfig {
+	edaClientV1Config := eigendaflags.ReadConfig(ctx)
+	edaClientV2Config := eigendaflags_v2.ReadConfig(ctx)
+
+	cfg := ProxyConfig{
+		ServerConfig: server.Config{
+			DisperseV2: edaClientV2Config.Enabled,
+			Host:       ctx.String(ListenAddrFlagName),
+			Port:       ctx.Int(PortFlagName),
+		},
+		EdaV1ClientConfig:   edaClientV1Config,
+		EdaV1VerifierConfig: verify.ReadConfig(ctx, edaClientV1Config),
+		PutRetries:          ctx.Uint(eigendaflags.PutRetriesFlagName),
+		MemstoreEnabled:     ctx.Bool(memstore.EnabledFlagName),
+		MemstoreConfig:      memstore.ReadConfig(ctx),
+		StorageConfig:       store.ReadConfig(ctx),
+		EigenDAV2Enabled:    edaClientV2Config.Enabled,
+	}
+
+	return cfg
+}
+
+// Check ... verifies that configuration values are adequately set
+func (cfg *ProxyConfig) Check() error {
+	if !cfg.MemstoreEnabled {
+		if cfg.EdaV1ClientConfig.RPC == "" {
+			return fmt.Errorf("using eigenda backend (memstore.enabled=false) but eigenda disperser rpc url is not set")
+		}
+	}
+
+	// provide dummy values to eigenda client config. Since the client won't be called in this
+	// mode it doesn't matter.
+	if cfg.MemstoreEnabled {
+		cfg.EdaV1ClientConfig.SvcManagerAddr = "0x0000000000000000000000000000000000000000"
+		cfg.EdaV1ClientConfig.EthRpcUrl = "http://0.0.0.0:666"
+	}
+
+	if !cfg.MemstoreEnabled {
+		if cfg.EdaV1ClientConfig.SvcManagerAddr == "" {
+			return fmt.Errorf("service manager address is required for communication with EigenDA")
+		}
+		if cfg.EdaV1ClientConfig.EthRpcUrl == "" {
+			return fmt.Errorf("eth prc url is required for communication with EigenDA")
+		}
```

Comment:
move these above, you already have another `if !cfg.MemstoreEnabled` cobnditional

Reply:
[20944e3](https://github.com/Layr-Labs/eigenda-proxy/pull/296/commits/20944e32884209950be9ff3c4ef017945ca3fc94)